File: java/src/main/java/io/fivexer/sdk/model/WorkerMetrics.java
package io.fivexer.sdk.model;

/**
 * One worker's recorded time and throughput: today, plus a rolling window.
 *
 * <p>The operator-plane mirror of what the worker sees in their own portal — the same numbers on
 * both planes, deliberately, since worked time comes from the one shift log.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkerMetrics {
    private String workerId;
    private WorkerMetricsDay today;
    private WorkerMetricsPeriod window;

    public String getWorkerId() { return workerId; }
    public WorkerMetricsDay getToday() { return today; }
    public WorkerMetricsPeriod getWindow() { return window; }
}

File: php/src/Model/WorkerMetrics.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * One worker's recorded time and throughput: today, plus a rolling window.
 *
 * The operator-plane mirror of what the worker sees in their own portal — the same numbers on
 * both planes, deliberately, since worked time comes from the one shift log.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkerMetrics
{
    public function __construct(
        public readonly string $workerId,
        public readonly WorkerMetricsDay $today,
        public readonly WorkerMetricsPeriod $window,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            workerId: (string) ($data['workerId'] ?? ''),
            today: WorkerMetricsDay::fromArray($data['today'] ?? []),
            window: WorkerMetricsPeriod::fromArray($data['window'] ?? []),
        );
    }
}
